src/ResponsiveProvider.js

```diff
@@ -3,49 +3,40 @@ import PropTypes from 'prop-types';
 import React, { useEffect, useMemo, useRef, useState } from 'react';
 import ResponsiveContext from './ResponsiveContext';
 
-const ResponsiveProvider = ({
-    initialMediaType,
-    defaultOrientation,
-    children,
-    breakpoints,
-    breakpointsMax,
-    mediaQueries,
-}) => {
-    const [isCalculated, setIsCalculated] = useState(false);
-    const [currentBreakpoint, setCurrentBreakpoint] = useState(
-        initialMediaType
-    );
+const useOrientation = (defaultOrientation) => {
```

Comment:
Hmmm, didn't think about this, that's a good point.  I see it as 

Advantages:

 - Smaller file, leaving room to focus on the main logic

Disadvantages:

 - may not be clear that these hooks are only used by `ResponsiveProvider` unless I put them in a subdirectory of `ResponsiveProvider`, keeping it in the same file makes this clear, though we would have to export them anyway if we'd like to test them as a separate unit.

Did I miss anything?  Overall though, I think it's a good idea.  If you still think I should separate it, where do you think is a good place for the `hooks` directory.

Reply:
I see what you are saying, what about creating a folder `ResponsiveProvider` and inside you would have `ResponsiveProvider.js` renamed to `index.js` and those 3 hooks could be directly inside that folder as well.

This would point them as "coupled to ResponsiveProvider" but when viewing the main code you could ignore them unless needed

Reply:
I also prefer the approach @SoaresMG suggested

Reply:
Nice suggestion, changed.  What do you guys think?